src/Features/CSharp/Portable/ConvertProgram/ConvertProgramTransform_ProgramMain.cs

```diff
@@ -56,8 +63,15 @@ public static async Task<Document> ConvertToProgramMainAsync(Document document,
             return document;
         }
 
-        private static async Task<SyntaxNode> GenerateProgramClassAsync(
+        private static bool IsProgramClass(ClassDeclarationSyntax declaration)
+        {
+            return declaration.Identifier.Text == WellKnownMemberNames.TopLevelStatementsEntryPointTypeName &&
```

Comment:
```suggestion
            return declaration.Identifier.ValueText == WellKnownMemberNames.TopLevelStatementsEntryPointTypeName &&
```

Reply:
Just curious, what is the difference between these 2 properties?

Reply:
@DoctorKrolic if you have `class @Program` for example, then `Text` is the actual source code text which is `@Program` while `ValueText` is just `Program`.

Also if you have `public class \u0050\u0072\u006F\u0067\u0072\u0061\u006D` (it might be surprising, but yes it's valid C# and is the same as `public class Program`)

Reply:
🤯